MusicBandSimulator/MusicBandSimulator/MusicBand.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MusicBandSimulator
+{
+    /// <summary>
+    /// клас Гурту. Містить конструктор Гурту, який приймає його ім'я. Має метод, який йому дозволяє грати. Містить метод для добавляння нових учасників гурту.
+    /// </summary>
+    internal class MusicBand
+    {
+        public string bandName { get; set; }
+
+        List<BandMember> bandMembers = new List<BandMember>();
+
+        public MusicBand(string bandName)
+        {
+            this.bandName = bandName;
+        }
+
+        public void Perform()
+        {
+            foreach (var bandMember in bandMembers)
+            {
+                bandMember.BandMemberPerform();
+            }
+        }
+
+        public void Add(BandMember newBandMember)
+        {
+            // умова, яка перевіряє, чи в Лісті є якісь об'єкти.
+            if (bandMembers.Count == 0) 
+            {
+                bandMembers.Add(newBandMember);
+            }
+            else
+            {
+                List<BandMember> bandMembers2 = new List<BandMember>();
+
+                //цикл що проходиться по колекції з усіма наявними об'єктами, перевіряючи, чи існують
+                //уже музиканти, що грають на інструменті нового музиканта, якого я хочу добавити в гурт.
+                //Добавляю всіх музикантів з унікальним інструментом до новоствореного Ліста. Якщо музикант
+                //з інструментом такий, який вже є в колективі - добавляю його в новий ліст. По завершенні
+                //циклу - надаю старому (основному Лісту значення нового ліста).
+                foreach (BandMember member in bandMembers)
```

Comment:
no the best solution from a memory performance perspective. Even considering that both `bandMembers` and `bandMembers2` won't be large collections having them both it's a waste of memory. 
think about how to use `List<T>.Remove(T)` method here.

Reply:
I was about to say the same, but then decided to let it slide as it's a lazy and interesting approach to solve the problem))
But you're the PO of this task, so it's up to you to decide)

Reply:
or you can just reassign specific list element

bandMembers[i] = newBandMember

Reply:
done